aten/src/ATen/native/sparse/cuda/SparseCUDAApplyUtils.cuh

```diff
@@ -351,6 +350,56 @@ __global__ void coalesceValuesKernel(
   }
 }
 
+// coalesceValuesKernel when Acctype is bool. Can be eliminated using
+// `if constexpr` when CUDA codes will be compiled under C++-17, see
+// gh-56055 for blockers.
+template <typename Dtype>
```

Comment:
I'd fully specialize this to bool. I don't think we want to accidentally accumulate into bool using any other type.

Reply:
Fixed.